mlrun/model_monitoring/api.py

```diff
@@ -242,6 +223,108 @@ def record_results(
     return model_endpoint
 
 
+def _model_endpoint_validations(
+    model_endpoint: ModelEndpoint,
+    model_path: str = "",
+    sample_set_statistics: typing.Dict[str, typing.Any] = None,
+    drift_threshold: float = None,
+    possible_drift_threshold: float = None,
+):
+    """
+    Validate that provided model endpoint configurations are matching the stored fields of the provided `ModelEndpoint`
+    object. Usually, this method is called by `get_or_create_model_endpoint()` in cases that the model endpoint is
+    already exist. If one of the validations fails, this method might raise an error, indicating on possible conflict.
+
+    :param model_endpoint:           A `ModelEndpoint` object that contains the expected values.
+    :param model_path:               Model store path. Should be similar to the `model_uri` that is stored under
```

Comment:
Should be similar to the `model_uri` that is stored under model_endpoint.spec.model_uri` =>
In case of endpoint_id reuse, should be similar to the `model_uri` that is stored under model_endpoint.spec.model_uri`